pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,394 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+__version__ = "0.1.1"
+
+import re
+import warnings
+from functools import cached_property
+from io import BytesIO
+from time import sleep
+from time import time as now
+
+import numpy as np
+from pyvisa import VisaIOError
+
+from pymeasure.adapters import PrologixAdapter
+from pymeasure.instruments import Instrument, RangeException, discreteTruncate
+
+
+class HP8753E(Instrument):
+    """Represents the HP8753E Vector Network Analyzer
+    and provides a high-level interface for taking scans of the
+    scattering parameters.
+
+    Template copied from Agilent 8722 class in pymeasure
+    """
+
+    SCAN_POINT_VALUES = [3, 11, 21, 26, 51, 101, 201, 401, 801, 1601]
+    SCATTERING_PARAMETERS = ["S11", "S12", "S21", "S22"]
+    S11, S12, S21, S22 = SCATTERING_PARAMETERS
+
+    start_frequency = Instrument.control(
+        "STAR?",
+        "STAR %e Hz",
+        """ A floating point property that represents the start frequency
+        in Hz. This property can be set.
```

Comment:
Regarding the docstrings see the [documentation](https://pymeasure.readthedocs.io/en/latest/dev/coding_standards.html#docstrings), especially:

Specific to properties, start them with “Control”, “Get”, “Measure”, or “Set” to indicate the kind of property, as it is not visible after import, whether a property is gettable (“Get” or “Measure”), settable (“Set”), or both (“Control”). In addition, it is useful to add type and information about [Restricting values with validators](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/properties.html#validators) (if applicable) at the end of the summary line, see the docstrings shown in examples throughout the [Adding instruments](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/index.html#adding-instruments) section. For example a docstring could be `"""Control the voltage in Volts (float strictly from -1 to 1)."""`.

Please note, that not all instruments in our library adhere to that rule yet.